test/bigquery/table.js

```diff
@@ -87,7 +86,23 @@ describe('BigQuery/Table', function() {
       useCleanCache: true,
       warnOnUnregistered: false
     });
+
     Table = require('../../lib/bigquery/table');
+
+    var tableCached = extend(true, {}, Table);
+
+    // Override all util methods, allowing them to be mocked. Overrides are
+    // removed before each test.
+    Object.keys(Table).forEach(function(tableMethod) {
+      if (typeof Table[tableMethod] !== 'function') {
+        return;
+      }
+
+      Table[tableMethod] = function() {
+        return (tableOverrides[tableMethod] || tableCached[tableMethod])
+          .apply(this, arguments);
+      };
+    });
```

Comment:
:+1: 

Side note (kinda) - do you think it's worth opening an issue to implement spies within our test? I think it would probably clean up the test files some and remove all the `objectOverride` stuff we have, since we should be able to easily clean them up in an `afterEach` hook.